openwisp_users/admin.py

```diff
@@ -271,6 +271,21 @@ def has_change_permission(self, request, obj=None):
             return False
         return super().has_change_permission(request, obj)
 
+    def has_delete_permission(self, request, obj=None):
+        if (
+            obj
+            and not obj.is_superuser
+            and request.user.is_staff
+            and not request.user.is_superuser
+            and 'Administrator' in request.user.groups.values_list('name', flat=True)
+        ):
+            org_owner = OrganizationOwner.objects.filter(
+                organization_user=OrganizationUser.objects.get(user=obj)
+            )
+            if not org_owner:
+                return True
+        return super().has_delete_permission(request, obj)
```

Comment:
I wouldn't do this.

Try with something like this (pseudocode):

```python
def delete_queryset(self, request, queryset):
    # change the queryset to join the table of organizationowner if you can
    for user in queryset:
        # if request.user is not super user
        # check if this user is an owner, if it is,
        # use self.message_user(request, '<ERROR MESSAGE>', messages.ERROR)
        # to add an error messsage and skip to the next loop cycle using continue 
        self.delete_model(request, user)
```

Please add some tests for this:

- single user delete, non superuser deletes owner, expects failure and ensure owner is not deleted
- single user delete, superuser deletes owner, expects success and ensure owner is deleted
- bulk delete, non super user deletes two users (one owner and one normal user), expects the error message, the owner not being deleted but the normal user being deleted
- bulk delete, super user deletes two users (one owner and one normal user), expects no error message, both users should be deleted

let me know if it's clear